### ISSUE TITLE
[Feature] Support Artemis the next generation of ActiveMQ

### ISSUE THREAD

**Issue by CzyerChen** [apache/skywalking#11830]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Since 2023, more and more people tried to use Artemis instead of ActiveMQ, and also two "flavors" of ActiveMQ available officially- the well-known "classic" broker and the "next generation" broker code-named Artemis. 
It inherits the core of ActiveMQ 5 but has been refactored and improved to make it more performant, scalable, and cloud-native.
[Apache ActiveMQ Artemis](https://activemq.apache.org/)

### Use case

If I work with Artemis, the producer and consumer can be traced.

### Are you willing to submit a pull request to implement this on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Apart from tracing, are you going to add server monitoring for this one? I am not sure how much different it would be, compared with ActiveMQ Classic.

**Comment by wu-sheng** [comment]
You have been involved in the agent tracing deeply, are you interested in expanding your tech field in this open-source community?

**Comment by CzyerChen** [comment]
> You have been involved in the agent tracing deeply, are you interested in expanding your tech field in this open-source community?

Yes, recently the last plugin in this part, then will try monitoring.

**Comment by CzyerChen** [comment]
> Apart from tracing, are you going to add server monitoring for this one? I am not sure how much different it would be, compared with ActiveMQ Classic.

Yes, I also want to do the ActiveMQ monitoring.

from two different clients 
|item|ActiveMQ-Classic|ActiveMQ-Artemis|
|--|--|--|
|client|org.apache.activemq.activemq-client|org.apache.activemq.artemis-jakarta-client|
|producer|org.apache.activemq.ActiveMQMessageProducer.send|org.apache.activemq.artemis.jms.client.ActiveMQMessageProducer.doSendx|
|consumer|org.apache.activemq.ActiveMQMessageConsumer.dequeue|org.apache.activemq.artemis.jms.client.ActiveMQMessageConsumer.getMessage|

**Comment by wu-sheng** [comment]
What do you mean `from two different clients`? My asking is like this https://skywalking.apache.org/docs/main/next/en/setup/backend/backend-rabbitmq-monitoring/. But ActiveMQ classic and Artemis are not supported yet.

**Comment by CzyerChen** [comment]
> What do you mean `from two different clients`? My asking is like this https://skywalking.apache.org/docs/main/next/en/setup/backend/backend-rabbitmq-monitoring/. But ActiveMQ classic and Artemis are not supported yet.

I plan to support tracing for Artemis firstly, and then plan to support ActiveMQ-classic monitoring feature.

**Comment by wu-sheng** [comment]
The thing I want to highlight is, tracing is agent thing, server side monitoring is more around OpenTelemetry(or something) integration at the server side.

**Comment by CzyerChen** [comment]
> The thing I want to highlight is, tracing is agent thing, server side monitoring is more around OpenTelemetry(or something) integration at the server side.

Yes, I understand.

**Comment by wankai123** [comment]
@CzyerChen Hi, thanks for your contribution, do you have any plan about `server side monitoring of ActiveMQ`?  I ask this because we used activeMQ in our [Showcase sample-service](https://github.com/apache/skywalking-showcase/blob/d738073a1ad0511cb37c66607ae100c6dd0662fa/services/songs-service/src/main/java/org/apache/skywalking/showcase/services/song/mq/SongMessageSender.java#L37).  Due to SkyWalking v10 support Hierarchy and Cross Layers topology and dashboards, we are optimizing the sample-service to make it could show full layers dashboard. 

If the `server side monitoring of ActiveMQ` will implement recently we' happy to reserve it in sample-service rather than switch to another MQ.
